mission/finite_state_machine/scripts/gate_choose_side.py

```diff
@@ -0,0 +1,160 @@
+from turtle import pos
+import rospy
+import smach
+from smach import StateMachine
+from geometry_msgs.msg import Pose, Point, Quaternion, Twist
+from std_msgs.msg import String
+from landmarks.srv import request_position
+import actionlib
+from vortex_msgs.msg import VtfPathFollowingAction, VtfPathFollowingGoal, SetVelocityGoal, SetVelocityAction, DpSetpoint
+from nav_msgs.msg import Odometry
+
+
+from tf.transformations import quaternion_from_euler
+from fsm_helper import dp_move, get_pose_in_front, rotate_certain_angle, get_pose_to_side
+from vortex_msgs.srv import ControlMode
+
+class GateSearch(smach.State):
+    def __init__(self):
+        smach.State.__init__(self, outcomes=['preempted', 'succeeded', 'aborted'])           
+        self.landmarks_client = rospy.ServiceProxy('send_positions',request_position)
+        rospy.wait_for_service('send_positions')
+        self.object = self.landmarks_client("gate").object
+
+        self.state_pub = rospy.Publisher('/fsm/state',String,queue_size=1)
+
+        vtf_action_server = "/controllers/vtf_action_server"
+        self.vtf_client = actionlib.SimpleActionClient(vtf_action_server, VtfPathFollowingAction)  
```

Comment:
There is a lot of boilerplate here for each state class, so it might be an idea to create a base class that the rest can inherit from. No need to do this in this PR, but is a point of cleanup for the future